dev/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -185,16 +208,48 @@ HRESULT ParseRootManifestFromXmlReaderInput(IUnknown* input)
 
 HRESULT ParseFileTag(IXmlReader* xmlReader)
 {
-    HRESULT hr = S_OK;
-    XmlNodeType nodeType;
     PCWSTR localName = nullptr;
     PCWSTR value = nullptr;
-    hr = xmlReader->MoveToAttributeByName(L"name", nullptr);
-    RETURN_HR_IF(HRESULT_FROM_WIN32(ERROR_SXS_MANIFEST_PARSE_ERROR), hr != S_OK);
-    RETURN_IF_FAILED(xmlReader->GetValue(&value, nullptr));
-    std::wstring fileName{ !value ? L"" : value };
-    RETURN_HR_IF(HRESULT_FROM_WIN32(ERROR_SXS_MANIFEST_PARSE_ERROR), fileName.empty());
+    std::wstring fileName;
     auto locale = _create_locale(LC_ALL, "C");
+    // Using this pattern intead of calling multiple MoveToAttributeByName improves performance
+    HRESULT hr = xmlReader->MoveToFirstAttribute();
```

Comment:
Presumably, since we change the way we parse things here, we could _possibly_ make a 1.4 app that was working fine fail at runtime when built with 1.5, correct?  I'm not saying I see such a problem, I'm just trying to make sure I understand the stakes.

Reply:
Yes, in theory